### PR TITLE
Create hackathons with geocoding

### PR DESCRIPTION
**Pull request by northeastprince**

### CODE REVIEW

File: db/migrate/20230711134548_create_hackathons.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class CreateHackathons < ActiveRecord::Migration[7.0]
+  def change
+    create_table :hackathons do |t|
+      t.string :name, null: false
+
+      t.integer :status, default: 0, null: false
+
+      t.datetime :starts_at
+      t.datetime :ends_at
```
Comment (garyhtou) [inline db/migrate/20230711134548_create_hackathons.rb:9]: Let's make these required!
`null: false`

**Review by garyhtou — CHANGES_REQUESTED** [review]
Amazing! There's just one comment about the address geocoding

File: test/models/hackathon/regional_test.rb
Comment (garyhtou) [inline test/models/hackathon/regional_test.rb:1]: great tests!

File: app/models/hackathon/regional.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module Hackathon::Regional
+  extend ActiveSupport::Concern
+
+  included do
+    geocoded_by :address
```
[the remark this answers is not in the document]
Comment (northeastprince) [reply app/models/hackathon/regional.rb:5]: The `address` is mainly just for user input, so the organizer enters the location however specific they want it (city + country, street + city + country, etc.), and it gets geocoded and reverse-geocoded, storing the full address retrieved with the lookup, as well as the different fields returned by the service (it sounds like you were suggesting this, but this implementation already stores those fields?). Although to leave this more open, we could also join those fields together `if address.blank?` for the initial geocoding process, making `Hackathon#create` a little more flexible.

For a `Subscription`, it doesn't have to be as specific, so for that I was just gonna have an `attr_accessor` for geocoding the country, province, and city, but that can be changed in the future PR for that.

[the remark this answers is not in the document]
Comment (northeastprince) [reply app/models/hackathon/regional.rb:5]:
```ruby
module Hackathon::Regional
  # ...
  def address
    super.presence || [city, province, country_code, postal_code].compact.join(", ")
  end
end

We can add something like that if you want. ^^^

**Review by garyhtou — APPROVED** [review]

File: app/models/hackathon/regional.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (garyhtou) [reply app/models/hackathon/regional.rb:5]: for posterity, we chatted about this on call and decided to opt for a new `street` column that will be used for user input (in addition to city, province, etc.). these fields will be overridden by reverse geocoding

**Pull request merged**